src/sqlfluff/core/config.py

```diff
@@ -294,17 +367,46 @@ def _incorporate_vals(ctx: dict, vals: List[ConfigElemType]) -> dict:
         return ctx
 
     @staticmethod
-    def _validate_configs(configs: Iterable[ConfigElemType], file_path):
+    def _validate_configs(
+        configs: Iterable[ConfigElemType], file_path
+    ) -> List[ConfigElemType]:
         """Validate config elements against removed list."""
-        for k, _ in configs:
-            if k in REMOVED_CONFIGS:
+        config_map = {cfg.old_path: cfg for cfg in REMOVED_CONFIGS}
+        new_configs = []
+        for k, v in configs:
+            if k in config_map.keys():
                 formatted_key = ":".join(k)
-                raise SQLFluffUserError(
-                    f"Config file {file_path} set an outdated config "
-                    f"value {formatted_key}.\n\n{REMOVED_CONFIGS[k]}\n\n"
-                    "See https://docs.sqlfluff.com/en/stable/configuration.html"
-                    " for more details."
-                )
+                removed_option = config_map[k]
+                # Is there a mapping option?
+                if removed_option.translation_func and removed_option.new_path:
+                    # Mutate and warn.
+                    v = removed_option.translation_func(v)
+                    k = removed_option.new_path
+                    formatted_new_key = ":".join(k)
+                    # NOTE: At the stage of emitting this warnig, we may not yet
+                    # have set up red logging because we haven't yet loaded the config
+                    # file. For that reason, this error message has a bit more padding.
```

Comment:
Should we consider adding some ASCII decoration as well? E.g.: `*********`, `======`, something like that.

Reply:
I think the newlines are enough for now - they space it out nicely.